alerting/src/main/kotlin/org/opensearch/alerting/util/DocLevelMonitorQueries.kt

```diff
@@ -378,25 +461,87 @@ class DocLevelMonitorQueries(private val client: Client, private val clusterServ
+        oldField.entries.forEach {
+            if (newField.containsKey(it.key)) {
+                if (it.value is Map<*, *> && newField[it.key] is Map<*, *>) {
+                    mergedField[it.key] =
+                        mergeConflictingFields(it.value as Map<String, Any>, newField[it.key] as Map<String, Any>)
+                } else {
+                    mergedField[it.key] = it.value
+                }
+            } else {
+                mergedField[it.key] = it.value
+            }
+        }
+
+        newField.entries.forEach {
+            if (!oldField.containsKey(it.key)) {
+                mergedField[it.key] = it.value
+            }
+        }
+        return mergedField
+    }
+
+    fun getAllConflictingFields(clusterState: ClusterState, concreteIndices: List<String>): Set<String> {
+        val conflictingFields = mutableSetOf<String>()
+        val allFlattenPaths = mutableMapOf<String, MutableMap<String, Any>>()
+        concreteIndices.forEach { concreteIndexName ->
+            if (clusterState.routingTable.hasIndex(concreteIndexName)) {
+                val indexMetadata = clusterState.metadata.index(concreteIndexName)
+                if (indexMetadata.mapping()?.sourceAsMap?.get("properties") != null) {
+                    val properties = (
+                        (indexMetadata.mapping()?.sourceAsMap?.get("properties"))
+                            as MutableMap<String, Any>
+                        )
+                    // Node processor function is used to process leaves of index mappings tree
+                    //
+                    val leafNodeProcessor =
+                        fun(fieldName: String, _: String, props: MutableMap<String, Any>): Triple<String, String, MutableMap<String, Any>> {
+                            return Triple(fieldName, fieldName, props)
+                        }
+                    // Traverse and update index mappings here while extracting flatten field paths
+                    val flattenPaths = mutableMapOf<String, MutableMap<String, Any>>()
+                    traverseMappingsAndUpdate(properties, "", leafNodeProcessor, flattenPaths)
+
+                    flattenPaths.forEach {
+                        if (allFlattenPaths.containsKey(it.key) && allFlattenPaths[it.key]!! != it.value) {
+                            conflictingFields.add(it.key)
+                        }
+                        allFlattenPaths.putIfAbsent(it.key, it.value)
+                    }
+                }
+            }
+        }
+        return conflictingFields
+    }
+
     /**
      * Adjusts max field limit index setting for query index if source index has higher limit.
      * This will prevent max field limit exception, when source index has more fields then query index limit
      */
-    private suspend fun checkAndAdjustMaxFieldLimit(sourceIndex: String, concreteQueryIndex: String) {
+    private suspend fun checkMaxFieldLimit(sourceIndex: String): Long {
```

Comment:
the java doc description for this method seems like it should describe `adjustMaxFieldLimitForQueryIndex`

Reply:
can we update the code comments for this method and below method

Reply:
updated it.